cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -123,6 +124,30 @@ leaderElection:
 		t.Fatal(err)
 	}
 
+	unknownFieldConfig := filepath.Join(tmpDir, "scheduler_invalid_unknown_field.yaml")
+	if err := ioutil.WriteFile(unknownFieldConfig, []byte(fmt.Sprintf(`
+apiVersion: kubescheduler.config.k8s.io/v1alpha1
+kind: KubeSchedulerConfiguration
+clientConnection:
+  kubeconfig: "%s"
+leaderElection:
+  leaderElect: true
+foo: bar`, configKubeconfig)), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
```

Comment:
can you ensure the error message mentions the invalid field somehow?

Reply:
I added part of the expected error messages in the test cases under `expectedError`